src/v/kafka/server/handlers/topics/validators.h

```diff
@@ -143,4 +143,68 @@ struct unsupported_configuration_entries {
     }
 };
 
+struct compression_type_validator_details {
+    using validated_type = model::compression;
+
+    static constexpr const char* error_message
+      = "Unsupported compression type ";
+    static constexpr const auto config_name = topic_property_compression;
+};
+
+struct compaction_strategy_validator_details {
+    using validated_type = model::compaction_strategy;
+
+    static constexpr const char* error_message
+      = "Unsupported compaction strategy ";
+    static constexpr const auto config_name
+      = topic_property_compaction_strategy;
+};
+
+struct timestamp_type_validator_details {
+    using validated_type = model::timestamp_type;
+
+    static constexpr const char* error_message = "Unsupported timestamp type ";
+    static constexpr const auto config_name = topic_property_timestamp_type;
+};
+
+struct cleanup_policy_validator_details {
+    using validated_type = model::cleanup_policy_bitflags;
+
+    static constexpr const char* error_message = "Unsupported cleanup policy ";
+    static constexpr const auto config_name = topic_property_cleanup_policy;
+};
+
+template<typename T>
```

Comment:
question, is this a place where a validator_details concept, something roughly like 
```
requires(T t) { 
    typename T::validated_type; 
    { t.config_name } ...
    { t.error_message } ...
}
```
would be useful?

Reply:
In terms of correctness I think it would be the same (i.e. fail at compile time if the details don't match the expectations) and I feel like it adds some boilerplate for something that's essentially quite simple. That being said, thanks for suggesting and I'd go for it if you feel strongly about it. Let me know.